src/Components/ServiceScene/Breakdowns/FilterByPatternsButton.tsx

```diff
@@ -17,15 +18,22 @@ export class FilterByPatternsButton extends SceneObjectBase<FilterByPatternsButt
       return;
     }
 
-    // remove from the other list if it's there
-    if (logExploration.state.patterns?.find((p) => p.pattern === this.state.pattern)) {
-      logExploration.setState({
-        patterns: logExploration.state.patterns.filter((pattern) => pattern.pattern !== this.state.pattern),
-      });
-    }
+    const { patterns = [] } = logExploration.state;
+
+    // Remove the pattern if it's already there
+    const filteredPatterns = patterns.filter((pattern) => pattern.pattern !== this.state.pattern);
+
+    // Analytics
+    const includePatternsLength = filteredPatterns.filter((p) => p.type === 'include')?.length ?? 0;
+    const excludePatternsLength = filteredPatterns.filter((p) => p.type === 'exclude')?.length ?? 0;
+    reportAppInteraction('service_selection', 'patterns_filtered', {
```

Comment:
Brainstorming: we could have some JSON somewhere where we could categorize and grab these event names, potentially with some explanation. For example:

```ts
const USER_EVENTS = {
  categories: {
    // Index Page
    service_selection: 'service_selection',
  },
  subcategories: {
    service_selection: {
      // Applying pattern filter
      patterns_filtered: 'patterns_filtered',
    }
   // ...
  }
}
```

```suggestion
    reportAppInteraction(USER_EVENTS.categories.service_selection, USER_EVENTS.subcategories.service_selection.patterns_filtered, {
```

With something like this, it should be very simple to know what's being instrumented, and create dashboards without having to go to each location to get the event names.

Reply:
Great idea! I'll do that and re-request your review.